### ISSUE TITLE
Documenting support for multiple plotting backends across HoloViz

### ISSUE THREAD

**Issue by maximlt** [holoviz/holoviz#322]
Some of the HoloViz libraries like HoloViews or hvPlot can generate plots for different plotting backends (currently Bokeh, Matplotlib and Plotly). HoloViews' [Reference Gallery](https://holoviews.org/reference/index.html) has a toggle in the gallery page that allows to select a plotting backend.

<img width="1071" alt="image" src="https://user-images.githubusercontent.com/35924738/184362042-cc3b38f7-998f-4716-87cf-f8a9c39eb911.png">

Then each page has links to the element references for the other plotting backends.

<img width="525" alt="image" src="https://user-images.githubusercontent.com/35924738/184362259-cc7d6a5d-a9b8-4a99-aa9d-08d9d0cef58c.png">

In practice for this to work each reference page corresponds to a notebook. So for instance the `Scatter` element has 3 notebooks, one for each plotting backend.

Currently hvPlot doesn't have in place such system. Instead to let the users know that it is actually possible to use another plotting backend that the default one (Bokeh), the *Plotting* guide has been duplicated.

<img width="238" alt="image" src="https://user-images.githubusercontent.com/35924738/184362814-a267ae07-aa73-41f2-b2e0-908f70ae7139.png">

In terms of documentation maintenance, HoloViews' solution isn't quite ideal, as any documentation change has to be made to three different notebooks. An advantage of this solution is that users can download a notebook that is going to run with the right backend. The situation of hvPlot is also far from being great, with clearly not enough ways to let the users know about the support of different plotting backends.

An effect of the difficulty to display the support for different backends is that the documentation ultimately tend to lean towards one backend, and that will be Bokeh for both HoloViews and hvPlot. Had there been an easy option to display the plots at the same time, e.g. in tabs as is often done with code snippets, today people might be using more the other backends (and we'd feel more embarrassed for some of the lack of support they have 🙃 ).

Ideally we would find a solution that could be applied to all the relevant websites (GeoViews too maybe?). I assume it should account for:

* UX/UI: should look nice and be easy to use. Additional UX question to ask would for instance be, how important is it for the references/guides to be downloadable as notebooks (as the solution might affect the downloaded content)?
* Maintenance: maintaining HoloViews' reference notebooks is a bit of a pain because of the duplication, the solution should try to aim at improving that, making it worse or expanding that would be a no-go

**Comment by jbednar** [comment]
I agree with all of this, and I anxiously await someone suggesting a good solution!
